tests/README.md

```diff
@@ -88,28 +88,27 @@ Run using
 This test compares the stdlib stubs against the objects at runtime. Because of
 this, the output depends on which version of Python and on what kind of system
 it is run.
-Thus the easiest way to run this test is via Github Actions on your fork;
-if you run it locally, it'll likely complain about system-specific
-differences (in e.g, `socket`) that the type system cannot capture.
+As such, if you run this test locally, it *may* complain about system-specific
```

Comment:
The original text is more prescriptive about how you should run stubtest, which I think is valuable (I don't think it's a particularly worthwhile or even feasible endeavour to make test_stubtest pass cleanly on all possible configurations of python and system). Last time I checked it was "likely" that you'd get system-specific complaints. What is the goal of this change?

Reply:
Well, I've run `stubtest_stdlib.py` many times locally, and never once run into an issue where stubtest complained about something locally but didn't in CI. So whether or not it's "likely" that you'll get system-specific complaints would appear itself to be system-specific :)

Reply:
> the type system genuinely "cannot capture" most extant system-specific differences.

I don't think there are _that_ many allowlist entries left that are due to platform-specific differences. And I think there's a solid chance that we could remove a lot of the remaining ones, if somebody has the inclination to put in the necessary work.

Reply:
> whether or not it's "likely"

Oh okay, yeah that makes sense. For reference, when I run `stubtest_stdlib.py` I get: `Found 139 errors (checked 485 modules)`.

> I don't think there are that many allowlist entries left 

The presence of allowlist entries is fine — by definition those don't cause `stubtest_stdlib.py` to fail. The issue is all the random ways in which local system and local Python can differ from CI, causing `stubtest_stdlib.py` to fail when run locally.

These can be worked around by adding regexes to stubtest allowlists (which is what we ended up doing with `socket`) and using the allow-unused-empty-regex trick, but I don't think this work is particularly worthwhile and can result in false negatives. For a dumb example, if someone has a Python build that doesn't include tkinter, there just isn't a great way to avoid stubtest tkinter errors

More generally, there's a norm that if you checkout a project and run its tests, and those tests fail, something unusual is happening. This just isn't the case with stubtest. Fwiw, the wording here was introduced because this confused some contributors, e.g. https://github.com/python/typeshed/issues/3782#issuecomment-592438089

Reply:
> which is what we ended up doing with `socket`

Only on linux these days ;)

Cool, this all makes sense, I'll push a commit to improve this bit

Reply:
> For a dumb example, if someone has a Python build that doesn't include tkinter, there just isn't a great way to avoid stubtest tkinter errors

Yeah, this is definitely true. I guess what I'm objecting to here a little bit is "the _type_ system cannot capture" this stuff. For a lot of this stuff, it's not really the _type_ system (we could probably describe differences between various linux systems if we were prepared to do stuff like `if sys.platform == "freebsd"` or whatever). For that kind of thing, it's more "we don't want to get into all of that because it's a fool's errand to try to capture platform-specific minutiae for little-used and unusual platforms".